internal/jobs/launch_instance_aws.go

```diff
@@ -308,20 +308,27 @@ func FetchInstancesDescriptionAWS(ctx context.Context, args *LaunchInstanceAWSTa
 			span.SetStatus(codes.Error, "cannot get list instances description")
 			return fmt.Errorf("cannot get list instances description: %w", errRetry)
 		}
+		logger.Trace().Msgf("AWS returned %d instance details", len(instancesDescriptionList))
 
 		if len(instancesDescriptionList) == 0 {
 			return ErrTryAgain
 		}
 
-		for _, instance := range instancesDescriptionList {
+		for i, instance := range instancesDescriptionList {
+			logger.Trace().Msgf("Instance id %s private IPv4:%s public IPv4:%s DNS:%s",
+				instancesIDList[i],
```

Comment:
!. Are we sure that the instancesIDList would have the same order as instancesDescriptionList? :thinking: 
instancesDescriptionList comes from a call of ec2Client.DescribeInstanceDetails. It parses a response from ec2 client, right?
For example, from the description of the DescribeInstances function:
> If you specify an instance that you do not own, it is not included in the output.
If we have had instancesIDList with such ID, this would probably break the order.

2. We are iterating instancesIDList within a loop to iterate through instancesDescriptionList. In theory, based on the upper quote, it should be true that
`len(instancesIDList) >  len(instancesDescriptionList)`
But I guess we are playing with fire here.

Reply:
Suggestion: Let's use instance.ID here, if that is possible.